addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -329,6 +338,11 @@ export class WebsitePreview extends Component {
     }
 
     _onPageLoaded() {
+        if (this.lastHiddenPageURL !== this.iframe.el.contentWindow.location.href) {
+            // Hide Ace Editor when moving to another page.
+            this.websiteService.context.showAceEditor = false;
+            this.lastHiddenPageURL = undefined;
+        }
```

Comment:
The only case I found where `this.lastHiddenPageURL` and the current url are the same is when we are on a page (e.g. Contact Us) and we go to the same page by clicking on its menu. Are there any other cases ?
(If we refresh, then the previous url is undefined, and if we switch languages, then the links are different because of the "/language_code" in it)
Do we want to keep the editor open in these case ?
If not, then I think directly setting `showAceEditor` to `false` at each pageLoad without checking the URLs would be alright :wink:

Reply:
It is to keep the editor open on save, to make it possible for the user to see the impact of its current changes before continuing to do more changes.